### PR TITLE
Updated env to remove mscratchcsw

### PR DESCRIPTION
**Pull request by silabs-hfegran**

### CODE REVIEW

File: tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv
Comment (silabs-robin) [inline tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv:3002]: Nice. This seems safer.
Broad antecedent, stricter consequent.

File: tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv
Comment (silabs-robin) [inline tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv:1]: The PR is draft because we have to wait for RTL before merge, right?

File: tb/uvmt/uvmt_cv32e40x_imperas_dv_wrap.sv
Comment (silabs-robin) [inline tb/uvmt/uvmt_cv32e40x_imperas_dv_wrap.sv:493]: I remember the discussion about this CSR being removed.
So this looks good then.

File: tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv
Comment (silabs-robin) [inline tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv:2939]: This is x-dv, so isn't it already given that the core is cv32e40x?
Are we adding the conditional so it can be ported over to the S?

File: tests/uvmt/base-tests/uvmt_cv32e40x_base_test_constants.sv
Comment (silabs-robin) [inline tests/uvmt/base-tests/uvmt_cv32e40x_base_test_constants.sv:24]: I guess there might be some way to do this without the preprocessor, but it... seems innocuous?
Looks like the code is equally as clean as if one had used e.g. generate blocks. Looks good.

File: tests/programs/custom/clic/clic.c
Comment (silabs-robin) [inline tests/programs/custom/clic/clic.c:2186]: This is difficult for me to review, because I don't know clic very well, and because... assembly is difficult.
So I will simply trust it.
The code gets put to the test, so there is assurance in the self-checking.

File: tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv
Comment (silabs-hfegran) [reply tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv:2939]: Yes, I am trying to keep the largest amount of commonality between the tests, so that bugfixes and changes can be more easily ported between the cores.

File: tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv
Comment (silabs-hfegran) [reply tb/uvmt/uvmt_cv32e40x_clic_interrupt_assert.sv:1]: Correct

**Review by silabs-robin — APPROVED** [review]
Approving based on:
Comments in this PR.

**Comment by silabs-hfegran** [comment]
ci_check ok apart from iss mismatch in ci_check.

**Pull request merged**
